src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs

```diff
@@ -318,19 +320,27 @@ static int GetNdkApiLevel(string androidNdkPath, string androidApiLevel, Android
 					string seqpointsFile = Path.Combine(outdir, string.Format ("{0}.msym",
 						Path.GetFileName (assembly.ItemSpec)));
 
-					string aotOptions = string.Format (
-						"{0}--aot={9}{8}{1}outfile={2},asmwriter,mtriple={3},tool-prefix={4},ld-flags={5},llvm-path={6},temp-path={7}",
-						EnableLLVM ? "--llvm " : string.Empty,
-						AotMode != AotMode.Normal ? string.Format("{0},", AotMode.ToString().ToLowerInvariant()) : string.Empty,
-						QuoteFileName (outputFile),
-						mtriple,
-						QuoteFileName (toolPrefix),
-						ldFlags,
-						QuoteFileName (SdkBinDirectory),
-						QuoteFileName (outdir),
-						sequencePointsMode == SequencePointsMode.Offline ? string.Format("msym-dir={0},", QuoteFileName(outdir)) : string.Empty,
-						AotAdditionalArguments != string.Empty ? string.Format ("{0},", AotAdditionalArguments) : string.Empty
-					);
+					List<string> aotOptions = new List<string> ();
+
+					if (!string.IsNullOrEmpty (AotAdditionalArguments))
+						aotOptions.Add (AotAdditionalArguments);
+					if (sequencePointsMode == SequencePointsMode.Offline)
+						aotOptions.Add ("msym-dir=" + QuoteFileName (outdir));
+					if (AotMode != AotMode.Normal)
+						aotOptions.Add (AotMode.ToString ().ToLowerInvariant ());
+
+					aotOptions.Add ("outfile="     + QuoteFileName (outputFile));
+					aotOptions.Add ("asmwriter");
+					aotOptions.Add ("mtriple="     + mtriple);
+					aotOptions.Add ("tool-prefix=" + QuoteFileName (toolPrefix));
+					aotOptions.Add ("ld-flags="    + ldFlags);
+					aotOptions.Add ("llvm-path="   + QuoteFileName (SdkBinDirectory));
+					aotOptions.Add ("temp-path="   + QuoteFileName (outdir));
+
+					string aotOptionsStr = (EnableLLVM ? "--llvm " : "") + "--aot=" + string.Join (",", aotOptions);
+
+					if (!string.IsNullOrEmpty (ExtraAotOptions))
+						aotOptionsStr += (aotOptions.Count > 0 ? "," : "") + ExtraAotOptions
```

Comment:
My reasoning was because it's AOT options, there is no need to split them and reconstruct them right after. And you also point a very valid point where that could break it.